socket/src/commonMain/kotlin/mqtt/socket/TCPServer.kt

```diff
@@ -21,7 +21,14 @@ class TCPServer (val host: String, val port: UShort, val process: ServerProcess)
     suspend fun getClientConnection() {
         listen().collect {
             if (it != null) {
-                process.newInstance().startProcessing(it)
+                try {
+                    process.newInstance().startProcessing(it)
+                } catch (e: Exception) {
+                    //any exception bubbles to this will be caught here and ignored.
+                    //this will ensure other processing in the coroutine scope will not be impacted.
```

Comment:
should we atleast print the exception or call into a function pointer? I'm sure it would be great to log and trace what happened rather than throwing the exception away at this point

Reply:
Yes, I agree we should print a message on what happened. But we do not have a logging messaging, At this this should we go with a print message to stdout?

Reply:
what if we try something like:
```
suspend fun getClientConnection(socketExceptionCallback: ((Exception) -> Unit)? = null) {
        listen().collect {
            if (it != null) {
                process.newInstance().startProcessing(it)
                try {
                    process.newInstance().startProcessing(it)
                } catch (e: Exception) {
                    socketExceptionCallback?.invoke(e)
```

Reply:
also i don't think we need to do the `if (it != null)` check

Reply:
This is a great idea.

So we define a generic exceptionCallback class and we can use it anywhere we want.

Reply:
made the changes as suggested